libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -292,134 +292,138 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 	if (_variable.annotation().type)
 		return;
 
+	if (!_variable.typeName())
+	{
+		// This can still happen in very unusual cases where a developer uses constructs, such as
+		// `var a;`, however, such code will have generated errors already.
```

Comment:
Never state anything without proof ;)
Can you use `m_errorReporter.hasErrors()` here?

Reply:
well yeah, I tried, but ReferencesResolver seems to be invoked within TypeChecker *before* the actual errors are being generated by constructs such as `var a;` - for example `var_empty_decl_1.sol` is reported properly by the TypeChecker, but the ReferenceResolver comes into play *earlier* (hence, I cannot check for hasErrors() just yet). Do you have any proposals what we could do?

Reply:
I think it's fine - TypeChecker has an assert there.